packages/hardhat-core/src/internal/util/bigint.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  BigIntLike,
+  bufferToBigInt,
+  bufferToHex,
+  toBuffer,
+} from "@ethereumjs/util";
+
+function min(x: bigint, y: bigint): bigint {
+  return x < y ? x : y;
+}
+
+function max(x: bigint, y: bigint): bigint {
+  return x > y ? x : y;
+}
+
+function isBigInt(x: unknown): x is bigint {
+  return typeof x === "bigint";
+}
+
+function divUp(x: bigint, y: bigint): bigint {
+  let result = x / y;
+
+  if (x % y !== 0n) {
+    result = result + 1n;
+  }
+
+  return result;
+}
+
+function cmp(a: bigint, b: bigint): number {
+  return a < b ? -1 : a > b ? 1 : 0;
+}
+
+function mapNumberToBigInt<T>(x: number | T): bigint | T {
+  if (typeof x === "number") {
+    return BigInt(x);
+  }
+
+  return x;
+}
+
+function mapBigIntToNumber<T>(x: bigint | T): number | T {
+  if (typeof x === "bigint") {
+    return Number(x);
+  }
+
+  return x;
+}
+
+function toWord(x: bigint | number): string {
+  return x.toString(16).padStart(64, "0");
+}
+
+function fromBigIntLike(x: BigIntLike): bigint {
+  if (typeof x === "bigint") {
+    return x;
+  }
+  if (typeof x === "number" || typeof x === "string") {
+    return BigInt(x);
+  }
+  if (Buffer.isBuffer(x)) {
+    return bufferToBigInt(x);
+  }
+
+  const exhaustiveCheck: never = x;
+  return exhaustiveCheck;
+}
+
+function toHex(x: number | bigint): string {
+  return bufferToHex(toBuffer(x));
+}
+
+export const BigIntUtils = {
```

Comment:
Why this instead of just exporting everything individually?

Reply:
I wanted to explicitly have `BigIntUtils.fooBar(...)`, at least in some places. But that can also be done with `import * as BigIntUtils` or similar, and I don't think we use this pattern anywhere else, so I'll change it.

Reply:
Fixed in https://github.com/NomicFoundation/hardhat/pull/3082/commits/159b61cc2b2eb572d0d56bcbc279f0749aa94e2f